### ISSUE TITLE
Option to disable image extraction

### ISSUE THREAD

**Issue by michaelballantyne** [8bitgentleman/roam-depot-tweet-extract#9]
I'm finding the image extraction behavior a little inconsistent, and I don't really want to clutter the file uploads in my graph anyway. A way of disabling image extraction altogether would be wonderful.

Thanks for the wonderful extension, by the way!

**Comment by 8bitgentleman** [comment]
I can definitely add something like that, what are you finding inconsistent about the current image extraction?

**Comment by michaelballantyne** [comment]
- It didn't extract the image in this tweet: https://twitter.com/ShriramKMurthi/status/1047286807757565953, but does extract the one here: https://twitter.com/htmlenergy/status/1666949290525171713
- When the Tweet Template includes {IMAGES} but the tweet has no images (or the extractor fails to find them, like with the ShriramKMurthi tweet), the text {IMAGES} appears in the result.

A couple other things I've noticed, not image related:

- There also seems to be a big variance in how long it takes between triggering the extraction and seeing the result. Sometimes it's very snappy and other times theres a 5 second wait or I just have to trigger it again.
- The effect of my change to the Tweet Template seems to be lost when I close and re-open Roam. I can see it there in the settings, but the behavior has reverted to the default template. To get mine to take effect I have to remove my customized template, then re-insert it.

**Comment by 8bitgentleman** [comment]
These should all be addressed in the latest version, just released on the depot. Some tweets just don't extract and there's nothing I can do but for the ones that do I've added a little spinner to show that the extraction is happening

**Comment by michaelballantyne** [comment]
Awesome! Feels a lot nicer now.

Two things left that I notice:

1. It seems that if I trigger the extraction too quickly after pasting the URL in to Roam, it doesn't work. I get the spinning wheel but it never extracts. I have to make sure to wait at least a second or two.
2. Now that twitter allows long tweets, only the first portion is extracted, with a ... at the end. For example with https://twitter.com/TrungTPhan/status/1690903872095899649

**Comment by 8bitgentleman** [comment]
1. Nothing I can do about that if really if things haven't synced to roam yet. Though be aware that sometimes it can take a little bit for the tweet extraction passthrough server to 'wake up' 
2. I keep running into this and there doesn't seem to be a real way to solve it on my end unfortunately

**Comment by michaelballantyne** [comment]
Yeah I've noticed some delay from waiting for the server to wake up, but it does seem like the issue of pasting and then triggering the extract too quickly is consistently reproducible independently of that.

I wonder if there'd be a way to fix it by using the .addPullWatch API method to register a watcher that would notice when the change is eventually committed to the graph. Probably more complicated than it's worth though.

Better feedback about failure might make it less of a problem---if the spinner goes away and the tweet hasn't been extracted, the user would have a better idea they should retry and aren't just waiting for the server to start.
